winpr/libwinpr/utils/ini.c

```diff
@@ -820,5 +803,66 @@ void IniFile_Free(wIniFile* ini)
 
 	free(ini->sections);
 	free(ini->buffer);
+	if (ini->fp)
+		fclose(ini->fp);
 	free(ini);
 }
+
+wIniFile* IniFile_Clone(const wIniFile* ini)
+{
+	if (!ini)
+		return NULL;
+
+	wIniFile* copy = IniFile_New();
+	if (!copy)
+		goto fail;
+
+	if (ini->fp)
+	{
+		copy->fp = fdopen(dup(fileno(ini->fp)), "r");
+		if (!copy->fp)
+			goto fail;
+	}
+
+	copy->lineLength = ini->lineLength;
+	if (ini->filename)
+	{
+		copy->filename = _strdup(ini->filename);
+		if (!copy->filename)
+			goto fail;
+	}
+	if (ini->buffer)
+	{
+		copy->buffer = _strdup(ini->buffer);
+		if (!copy->buffer)
+			goto fail;
+	}
+
+	copy->readOnly = ini->readOnly;
+
+	for (size_t x = 0; x < ini->nSections; x++)
+	{
+		const wIniFileSection* cur = ini->sections[x];
+		if (!cur)
+			goto fail;
+
+		wIniFileSection* scopy = IniFile_AddToSection(copy, cur->name);
+		if (!scopy)
+			goto fail;
+
+		for (size_t y = 0; y < cur->nKeys; y++)
+		{
+			const wIniFileKey* key = cur->keys[y];
+			if (!key)
+			{
```

Comment:
Nit: This is inconsistent with the other lines above, where we don't use braces one-liners.